plugins/modules/k8s_drain.py

```diff
@@ -0,0 +1,394 @@
+            continue
+
+        # Check replicated Pod
+        owner_ref = pod.metadata.owner_references
+        if not owner_ref:
+            unmanaged.append((pod.metadata.namespace, pod.metadata.name))
+        else:
+            for owner in owner_ref:
+                if owner.kind == "DaemonSet":
+                    daemonSet.append((pod.metadata.namespace, pod.metadata.name))
+                else:
+                    to_delete.append((pod.metadata.namespace, pod.metadata.name))
+
+    warnings, errors = [], []
+    if unmanaged:
+        pod_names = ','.join([pod[0] + "/" + pod[1] for pod in unmanaged])
+        if not force:
+            errors.append("cannot delete Pods not managed by ReplicationController, ReplicaSet, Job,"
+                          " DaemonSet or StatefulSet (use option force set to yes): {0}.".format(pod_names))
+        else:
+            # Pod not managed will be deleted as 'force' is true
+            warnings.append("Deleting Pods not managed by ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet: {0}.".format(pod_names))
+            to_delete += unmanaged
+
+    # mirror pods warning
+    if mirror:
+        pod_names = ','.join([pod[0] + "/" + pod[1] for pod in mirror])
+        warnings.append("cannot delete mirror Pods using API server: {0}.".format(pod_names))
+
+    # local storage
+    if localStorage:
+        errors.append("cannot delete Pods with local storage: {0}.".format(pod_names))
+
+    # DaemonSet managed Pods
+    if daemonSet:
+        pod_names = ','.join([pod[0] + "/" + pod[1] for pod in daemonSet])
+        if not ignore_daemonset:
+            errors.append("cannot delete DaemonSet-managed Pods (use option ignore_daemonset set to yes): {0}.".format(pod_names))
+        else:
+            warnings.append("Ignoring DaemonSet-managed Pods: {0}.".format(pod_names))
+    return to_delete, warnings, errors
+
+
+class K8sDrainAnsible(object):
+
+    def __init__(self, module):
+        from ansible_collections.kubernetes.core.plugins.module_utils.common import (
+            K8sAnsibleMixin, get_api_client)
+
+        self._module = module
+        self._k8s_ansible_mixin = K8sAnsibleMixin(module)
+        self._k8s_ansible_mixin.client = get_api_client(module=self._module)
+
+        self._k8s_ansible_mixin.module = self._module
+        self._k8s_ansible_mixin.argspec = self._module.argument_spec
+        self._k8s_ansible_mixin.check_mode = self._module.check_mode
+        self._k8s_ansible_mixin.params = self._module.params
+        self._k8s_ansible_mixin.fail_json = self._module.fail_json
+        self._k8s_ansible_mixin.fail = self._module.fail_json
+        self._k8s_ansible_mixin.exit_json = self._module.exit_json
+        self._k8s_ansible_mixin.warn = self._module.warn
+        self._k8s_ansible_mixin.warnings = []
+
+        self._api_instance = core_v1_api.CoreV1Api(self._k8s_ansible_mixin.client.client)
```

Comment:
Is there a reason we aren't using the dynamic client here? I'd prefer to keep things consistent throughout the code base even if it means a few extra lines of code.

Reply:
dynamic cannot be used here as it does not return owner references (for pod) which is mandatory to determine whether a pod is eligible for deletion/eviction. Also, we are working exclusively with the ``pod`` object so so there is no added value to use the dynamic client.